From oliver@neukum.org  Thu Jun 19 12:06:07 2008
From: Oliver Neukum <oliver@neukum.org>
Date: Tue, 17 Jun 2008 22:30:48 +0200
Subject: USB: fix cdc-acm resume()
To: greg kh <greg@kroah.com>, Linux USB <linux-usb@vger.kernel.org>
Message-ID: <200806172230.49447.oliver@neukum.org>
Content-Disposition: inline


cdc-acm has
- a memory leak in resume()
- will fail to reactivate the read code path if this is needed.
his corrects it by deleting the useless relict code.

Signed-off-by: Oliver Neukum <oneukum@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/class/cdc-acm.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/drivers/usb/class/cdc-acm.c
+++ b/drivers/usb/class/cdc-acm.c
@@ -1125,9 +1125,6 @@ static void stop_data_traffic(struct acm
 	for (i = 0; i < acm->rx_buflimit; i++)
 		usb_kill_urb(acm->ru[i].urb);
 
-	INIT_LIST_HEAD(&acm->filled_read_bufs);
-	INIT_LIST_HEAD(&acm->spare_read_bufs);
-
 	tasklet_enable(&acm->urb_task);
 
 	cancel_work_sync(&acm->work);
